lib/ui/window/platform_message_response_dart.cc

```diff
@@ -8,17 +8,52 @@
 
 #include "flutter/common/task_runners.h"
 #include "flutter/fml/make_copyable.h"
+#include "flutter/fml/trace_event.h"
 #include "third_party/tonic/dart_state.h"
 #include "third_party/tonic/logging/dart_invoke.h"
 #include "third_party/tonic/typed_data/dart_byte_data.h"
 
+static std::atomic<uint64_t> platform_message_counter = 1;
+
 namespace flutter {
+namespace {
+template <typename Callback, typename TaskRunner, typename Result>
+void PostCompletion(Callback&& callback,
+                    const TaskRunner& ui_task_runner,
+                    bool* is_complete,
+                    const std::string& channel,
+                    Result&& result) {
+  if (callback.is_empty()) {
+    return;
+  }
+  FML_DCHECK(!*is_complete);
+  *is_complete = true;
+  uint64_t platform_message_id = platform_message_counter.fetch_add(1);
+  TRACE_EVENT_ASYNC_BEGIN1("flutter", "PlatformChannel:ScheduleResult",
+                           platform_message_id, "channel", channel.c_str());
+  ui_task_runner->PostTask(fml::MakeCopyable(
```

Comment:
We seem to have dropped the idea of callback being empty here?

Reply:
Nope, it's on line 26 which matches the previous logic.